### ISSUE TITLE
Batch norm with training=true hangs for channels > 64

### ISSUE THREAD

**Issue by pglusacTT** [tenstorrent/tt-metal#30279]
### Describe the bug

Forge models contain a lot of batch norm training operations found on models.
The most important models and their shapes for batch norm can be found in the table below (shapes are NCHW).
Running batch norm training with shapes that have C > 64 hang.

Model Name | Operand
-- | --
resnet_resnet101_cv_image_cls_torchvision | 1x1024x14x14xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x128x28x28xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x128x56x56xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x2048x7x7xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x256x14x14xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x256x28x28xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x256x56x56xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x512x14x14xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x512x28x28xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x512x7x7xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x64x112x112xbf16
resnet_resnet101_cv_image_cls_torchvision | 1x64x56x56xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x128x112x112xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x128x56x56xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x256x28x28xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x256x56x56xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x512x14x14xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x512x28x28xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x64x112x112xbf16
unet_unet_cityscapes_cv_image_seg_osmr | 1x64x224x224xbf16
vgg_torchvision_vgg11_bn_cv_image_cls_torchvision | 1x128x112x112xbf16
vgg_torchvision_vgg11_bn_cv_image_cls_torchvision | 1x256x56x56xbf16
vgg_torchvision_vgg11_bn_cv_image_cls_torchvision | 1x512x14x14xbf16
vgg_torchvision_vgg11_bn_cv_image_cls_torchvision | 1x512x28x28xbf16
vgg_torchvision_vgg11_bn_cv_image_cls_torchvision | 1x64x224x224xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x1024x14x14xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x1024x7x7xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x128x56x56xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x2048x7x7xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x256x28x28xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x256x56x56xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x512x14x14xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x512x28x28xbf16
wide_resnet_wide_resnet50_2_cv_image_cls_torch_hub | 1x64x112x112xbf16

### Steps to reproduce the issue

Run the following:

```python
import torch
import ttnn

device = ttnn.open_device(device_id=0)

N = 1
C = 128
H = 28
W = 28

x = torch.randn(N, C, H, W, dtype=torch.bfloat16)

xt = ttnn.from_torch(x, dtype=ttnn.bfloat16, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG)
mean = torch.zeros(1, C, 1, 1, dtype=torch.bfloat16)
variance = torch.zeros(1, C, 1, 1, dtype=torch.bfloat16)
mean_t = ttnn.from_torch(mean, dtype=ttnn.bfloat16, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG)
variance_t = ttnn.from_torch(variance, dtype=ttnn.bfloat16, device=device, layout=ttnn.TILE_LAYOUT, memory_config=ttnn.DRAM_MEMORY_CONFIG)

print("Executing batch norm")
yt = ttnn.batch_norm(
    xt, 
    eps=1e-5, 
    momentum=0.1, 
    running_mean=mean_t, 
    running_var=variance_t, 
    weight=None, 
    bias=None, 
    training=True
)
print("Batch norm executed")

yt_t = ttnn.to_torch(yt, dtype=torch.bfloat16, device=device)
meant_t = ttnn.to_torch(mean_t, dtype=torch.bfloat16, device=device)
variancet_t = ttnn.to_torch(variance_t, dtype=torch.bfloat16, device=device)
print("Batch norm results moved to CPU")

print(yt_t)
print(meant_t)
print(variancet_t)

ttnn.close_device(device)
```

### Please complete the following environment information

- Metal commit cdd0c7d586bc259de8cb8e5e9a3c50da8dd4a540
- Architecture: Wormhole N150

**Comment by abogdanovicTT** [comment]
Also, in YOLOv8 training, stablehlo.batch_norm_training is not supported for channels ≥ 64.
Shapes for batch norm are below:

```
4x80x80x80xf32
4x80x40x40xf32
4x80x20x20xf32 
4x128x40x40xf32 
4x128x20x20xf32 
4x256x20x20xf32  
```

**Comment by skhatriTT** [comment]
Hey @fahmedchanTT can you please help triage this to the right person?

**Comment by fahmedchanTT** [comment]
@bbradelTT Can you please take a look and reassign out as necessary?

**Comment by bbradelTT** [comment]
Nothing jumps out.

@iwroszTT please investigate what is going on.

**Comment by iwroszTT** [comment]
I was able to reproduce the issue. Working on finding the root cause.

**Comment by vladimirjovanovicTT** [comment]
@bbradelTT @iwroszTT thanks for looking into this; this causes hangs in backward passes of 56 Forge models (attached) so this is the top issue for Forge training.
We can extract a more extensive op sweep if neeeded. 

[bn_train_models.csv](https://github.com/user-attachments/files/23186606/bn_train_models.csv)

**Comment by iwroszTT** [comment]
Hi @vladimirjovanovicTT, we found a problem in running stats calculations. The attached PR, currently in tests and code review, fixes that problem.

**Comment by iwroszTT** [comment]
The fix for hang has just been merged to tt-metal main as Commit [7c6c393](https://github.com/tenstorrent/tt-metal/commit/7c6c39392d8fe44cb2cfa9013e2bab161297d544).
I checked all inputs listed above on WH and a smaller but seemingly representative sample on BH. We have also added a new unit test to cover this issue.
